tests/ui/option_take_on_temporary.rs

```diff
@@ -0,0 +1,15 @@
+// run-rustfix
+
+fn main() {
+    println!("Testing non erroneous option_take_on_temporary");
+    let mut option = Some(1);
+    let _ = Box::new(move || option.take().unwrap());
+
+    println!("Testing non erroneous option_take_on_temporary");
+    let x = Some(3);
+    x.as_ref();
+
+    println!("Testing erroneous option_take_on_temporary");
+    let x = Some(3);
+    x.as_ref().take();
+}
```

Comment:
How about a test case like:

```
let x = Some(42);
let xref = x.as_ref();
let _ = xref.take();
```

Bonus question: Should that lint?

Reply:
I don't think so because `take()` is called on `xref`, which stores the result of `x.as_ref()`